core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -492,12 +493,24 @@ public void failTaskRemotely(TaskId taskId, Throwable failureCause)
             SqlStage sqlStage = requireNonNull(stages.get(taskId.getStageId()), () -> "stage not found: %s" + taskId.getStageId());
             sqlStage.failTaskRemotely(taskId, failureCause);
         }
+
+        public void logDebugInfo()
+        {
+            if (!log.isDebugEnabled()) {
+                return;
+            }
+            log.debug("SqlStages:");
+            stages.forEach((stageId, stage) -> {
+                log.debug("SqlStage %s: %s", stageId, stage.getDebugInfo());
+            });
+        }
     }
 
     private static class Scheduler
             implements EventListener
     {
         private static final int EVENT_BUFFER_CAPACITY = 100;
+        private static final Duration SCHEDULER_STALLED_DURATION_THRESHOLD = Duration.succinctDuration(5, MINUTES);
```

Comment:
let's ittroduce a constant for 1m in `eventQueue.poll(1, MINUTES);`
and add a comment that SCHEDULER_STALLED_DURATION_THRESHOLD accuracy is affected by the poll timeout